source/llassetgen-cmd/source/main.cpp

```diff
@@ -0,0 +1,226 @@
+#include <CLI11.h>
+#include <codecvt>
+#include <map>
+#include <ostream>
+
+#include <algorithms.h>
+#include <helpstrings.h>
+
+#include <llassetgen/Atlas.h>
+#include <llassetgen/FntWriter.h>
+#include <llassetgen/FontFinder.h>
+
+using namespace llassetgen;
+
+// all printable ascii characters, except for space
+constexpr char ascii[] =
+    "!\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
+
+std::u32string UTF8toUCS4(const std::string& str) {
+#if _MSC_VER >= 1900  // handle Visual Studio bug
+    std::wstring_convert<std::codecvt_utf8<uint32_t>, uint32_t> ucs4conv;
+    auto convStr = ucs4conv.from_bytes(str);
+    return std::u32string(reinterpret_cast<const char32_t*>(convStr.data()), convStr.size());
+#else
+    std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t> ucs4conv;
+    return ucs4conv.from_bytes(str);
+#endif
+}
+
+std::vector<Vec2<size_t>> sizes(const std::vector<Image>& images, unsigned int downsamplingRatio) {
+    std::vector<Vec2<size_t>> imageSizes(images.size());
+    std::transform(images.begin(), images.end(), imageSizes.begin(),
+                   [downsamplingRatio](const Image& img) { return img.getSize() / downsamplingRatio; });
+    return imageSizes;
+}
+
+std::pair<std::string, std::string> outNames(const std::string& outPath) {
+    std::string pathWithoutExtension;
+    if (outPath.substr(outPath.length() - 4) == ".png") {
+        pathWithoutExtension = outPath.substr(0, outPath.length() - 4);
+    } else {
+        pathWithoutExtension = outPath;
+    }
+    return std::make_pair(pathWithoutExtension + ".png", pathWithoutExtension + ".fnt");
+}
+
+std::set<unsigned long> makeGlyphSet(const std::string& glyphs, const std::vector<unsigned int>& charCodes,
+                                     bool includeAscii) {
+    std::set<unsigned long> set;
+    for (const auto c : UTF8toUCS4(glyphs)) {
+        set.insert(c);
+    }
+
+    for (const auto c : charCodes) {
+        set.insert(c);
+    }
+
+    if (includeAscii) {
+        const char* p = ascii;
+        while (*p) {
+            set.insert(static_cast<unsigned long>(*p++));
+        }
+    }
+    return set;
+}
+
+int parseAtlas(int argc, char** argv) {
+    // Example: llassetgen-cmd atlas -d parabola --ascii -f Verdana atlas.png
+    CLI::App app{atlasHelp};
+
+    // positional arguments
```

Comment:
"positional" like in "file path" ? 
I think "positional" might also sound like some x,y-values....

Reply:
I use "positional" like the CLI11 lib, I don't know if it's a common term. It means the position of the argument matters, e.g. `cp here there` does something different than `cp there here`. Whereas `cp -r here there` is the same as cp here -r there` because `-r` is not positional.
A positional is basically any argument that doesn't have `--something` or `-s` in front of it ^^

Reply:
ah, that's good to know. Then that's fine as it is.